app/jobs/submit_iplc_listener_job.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+##
+# Background job to send requests to IPLC.
+# If Reshare has the item and we succesfully request it, let the user know it is on the way.
+# If Reshare does not have it or we cannot succesfully request it, trigger the normal Symphony request workflow.
+class SubmitIplcListenerJob < ApplicationJob
+  discard_on ActiveRecord::RecordNotFound do |job, _error|
+    Honeybadger.notify(
+      "Attempted to call Reshare for Request with ID #{job.request_id}, but no such Request was found."
+    )
+  end
+
+  def perform(request_id, instance_uuid)
+    request = Request.find(request_id)
+
+    Sidekiq.logger.info("Started SubmitIplcRequestJob for request #{request_id}")
+
+    begin
+      make_iplc_request(request, instance_uuid)
+    rescue StandardError => e
+      Honeybadger.notify("IPLC Request failed for #{request_id} with #{e}. Submitted to Symphony instead.")
+
+      request.send_to_symphony_now!
+    end
+
+    Sidekiq.logger.info("Completed SubmitIplcRequestJob for request #{request_id}")
+  end
+
+  def make_iplc_request(request, instance_uuid)
+    iplc_response = IplcWrapper.new(request, instance_uuid)
+
+    raise 'IPLC HTTP request failed' unless iplc_response.success?
+
+    request.borrow_direct_response_data = iplc_response.as_json
+    request.via_borrow_direct = true
+    request.save
+
+    request.send_approval_status!
+  end
+
+  # Basic client for working with the IPLC listener HTTP API.
+  class IplcWrapper
+    attr_reader :request, :instance_uuid
+
+    def initialize(request, instance_uuid)
+      @request = request
+      @instance_uuid = instance_uuid
+    end
+
+    def as_json(_options)
+      {
+        response: response,
+        params: iplc_params
+      }
+    end
+
+    def success?
+      iplc_request.status.success?
+    end
+
+    private
+
+    def iplc_request
+      @iplc_request ||= HTTP.get(Settings.borrow_direct.iplc_listener_url, params: iplc_params)
+    end
+
+    def response
+      @response ||= JSON.parse(iplc_request.body.to_s)
+    end
+
+    def iplc_params
+      {
+        svc_id: 'json',
+        req_id: iplc_patron_id,
+        rft_id: instance_uuid,
+        'res.org': 'ISIL:US-CST',
+        'svc.pickupLocation': iplc_pickup_location_code,
+        rfr_id: request.to_global_id.to_s
+      }
+    end
+
+    def iplc_patron_id
+      request.user.patron.university_id
```

Comment:
Does a patron always have a university_id?  I noticed that @corylown is testing for this in https://github.com/sul-dlss/mylibrary/pull/761/files/42f79639511ea51e2728fc2239d181bd2f7fd3c9..d21ea15dc62135fa44d1fb7c64a409f3f9f51d37#diff-12bc95bc120bd5dd5050c084ac44aaefea6a87f3bf01b373720a3e79e2ad1abeR21 and I wonder if it's the same value?